pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -109,12 +148,15 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 						// add to cache if the pod is running or pending
 						pc.update(ip, key)
 					}
-
+					if !IsPodReady(pod) {
```

Comment:
I am wondering if this util can replace the `v1.PodPending, v1.PodRunning:` check

Reply:
could not